org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -268,6 +268,7 @@ public IStatus run(IProgressMonitor monitor) {
 					workspaceDiagnosticsHandler.publishDiagnostics(monitor);
 					workspaceDiagnosticsHandler.addResourceChangeListener();
 					pm.registerWatchers();
+					pm.registerListeners();
```

Comment:
The register of listening to the change of `ReferencedLibraries` is delayed to `initialized` phase to avoid redundant updating classpath & registering watcher during initialization.